File: editor/en/[121]Best Time to Buy and Sell Stock.java
121
        Best Time to Buy and Sell Stock
        2023-02-24 16:06:43

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int maxProfit(int[] prices) {
        int cost = Integer.MAX_VALUE;
        int profit = 0;

        for (int price : prices) {
            cost = Math.min(cost, price);
            profit = Math.max(profit, price - cost);
        }

        return profit;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[121]Best Time to Buy and Sell Stock.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let cost = Infinity;
    let profit = 0;
    for (const price of prices) {
        cost = Math.min(cost, price);
        profit = Math.max(profit, price - cost);
    }
    return profit;
};
//leetcode submit region end(Prohibit modification and deletion)
